dask_glm/algorithms.py

```diff
@@ -1,155 +1,68 @@
-                                                     Xbeta, Xstep,
-                                                     y, func,
-                                                     backtrackMult=backtrackMult,
-                                                     armijoMult=armijoMult,
-                                                     stepSize=stepSize)
-
-        beta, stepSize, Xbeta, gradient, lf, func, step, Xstep = persist(
-            beta, stepSize, Xbeta, gradient, lf, func, step, Xstep)
-
-        stepSize, lf, func, step = compute(stepSize, lf, func, step)
-
-        beta = beta - stepSize * step  # tiny bit of repeat work here to avoid communication
-        Xbeta = Xbeta - stepSize * Xstep
-
-        if stepSize == 0:
-            print('No more progress')
-            break
-
-        # necessary for gradient computation
-        eXbeta = exp(Xbeta)
-
-        yk = -gradient
-        sk = -stepSize * step
-        stepSize *= stepGrowth
-
-        if stepSize == 0:
-            print('No more progress')
-
-        df = lf - func
-        df /= max(func, lf)
-        if df < tol:
-            print('Converged')
-            break
-
-    return beta
-
-
-def loglike(Xbeta, y):
-    #        # This prevents overflow
-    #        if np.all(Xbeta < 700):
-    eXbeta = exp(Xbeta)
-    return log1p(eXbeta).sum() - dot(y, Xbeta)
+from dask_glm.utils import dot, exp, log1p, absolute, sign
+from dask_glm.families import Logistic
 
 
-def compute_stepsize(beta, step, Xbeta, Xstep, y, curr_val, stepSize=1.0,
-                     armijoMult=0.1, backtrackMult=0.1):
+def compute_stepsize_dask(beta, step, Xbeta, Xstep, y, curr_val,
+                          loglike=Logistic.loglike, stepSize=1.0,
+                          armijoMult=0.1, backtrackMult=0.1):
+    beta, step, Xbeta, Xstep, y, curr_val = persist(beta, step, Xbeta, Xstep, y, curr_val)
     obeta, oXbeta = beta, Xbeta
+    (step,) = compute(step)
     steplen = (step ** 2).sum()
     lf = curr_val
     func = 0
     for ii in range(100):
         beta = obeta - stepSize * step
-        if ii and np.array_equal(beta, obeta):
+        if ii and (beta == obeta).all():
             stepSize = 0
             break
-        Xbeta = oXbeta - stepSize * Xstep
 
+        Xbeta = oXbeta - stepSize * Xstep
         func = loglike(Xbeta, y)
-        df = lf - func
+        Xbeta, func = persist(Xbeta, func)
+
+        df = lf - compute(func)[0]
         if df >= armijoMult * stepSize * steplen:
             break
         stepSize *= backtrackMult
 
     return stepSize, beta, Xbeta, func
 
-
-def compute_stepsize_dask(beta, step, Xbeta, Xstep, y, curr_val, stepSize=1.0,
-                          armijoMult=0.1, backtrackMult=0.1):
-    beta, step, Xbeta, Xstep, y, curr_val = persist(beta, step, Xbeta, Xstep, y, curr_val)
+def compute_stepsize(beta, step, Xbeta, Xstep, y, curr_val, loglike=Logistic.loglike,
+                     stepSize=1.0, armijoMult=0.1, backtrackMult=0.1):
```

Comment:
It's a genuine question, not a request.  It might be simpler to always pass around and use the one thing.  I don't know though.

`compute_stepsize` should no longer be used anywhere I think